scripts/Engineering/gui/engineering_diffraction/tabs/fitting/data_handling/data_presenter.py

```diff
@@ -234,12 +232,10 @@ def _update_plotted_ws_with_sub_state(self, ws_name, is_sub):
             self.plotted.add(ws_name)
 
     def _handle_selection_changed(self):
-        rows = self.view.get_selected_rows()
-        enabled = False
-        for row in rows:
-            if self.view.get_item_checked(row, 3):
-                enabled = True
-        self._enable_inspect_bg_button(enabled)
+        enable = True
+        if self.view.get_selected_rows() == {}:
```

Comment:
Opinion may be divided about this but I would say it's more pythonic to cast this to a bool (empty containers and `None` are always interpreted as `False`) - i.e. L236 could be replaced by
`if not self.view.get_selected_rows():`
However, you might argue this is less precise/more ambiguous, but it does allow one to change the data structure of whatever is returned by `self.view.get_selected_rows()` without breaking code like this.
In fact you could replace the innards of this function with a single line
`self._enable_inspect_bg_button(bool(self.view.get_selected_rows()))`
but this is less readable, so I'm happy with the above.

Reply:
Sorry @DanielMurphy22 - I made this change and discovered the `TwoWayRowDict` had no `__bool__ ` method so presumably that's why you had `=={}`! Since I'd already made the change I have added a `__bool__ ` method so it behaves more like a standard container.

Reply:
Yes @RichardWaiteSTFC you've spotted the problem I had! Your solution is much nicer. Thanks for tidying up this PR!